src/components/UserSelector.tsx

```diff
@@ -1,16 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cl from 'classnames';
+import { getUsers } from '../api/users';
+import { User } from '../types/User';
+import { Loader } from './Loader';
+
+type Props = {
+  currentUser: User | null;
+  setCurrentUser: (u: User | null) => void;
+  hasError: boolean;
+  setHasError: (h: boolean) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  currentUser,
+  setCurrentUser,
+  hasError,
+  setHasError,
+}) => {
+  const [users, setUsers] = useState<User[]>();
+  const [isFocusDropdown, setIsFocusDropdown] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getUsers()
+      .then(usersFromServer => {
+        setUsers(usersFromServer);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
```

Comment:
Fix it 
<img width="1202" alt="image" src="https://github.com/mate-academy/react_dynamic-list-of-posts/assets/64750958/c37efeb7-5fb2-4cd8-b70a-de16d2305b97">

Reply:
seriously? put the function in the dependency mass? or delete this array so that requests for users are constantly executed?
![image](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/103579437/b29a0f21-8972-4233-a870-eda213bdfb0a)